src/complex/Utilities/OStreamUtilities.cpp

```diff
@@ -0,0 +1,546 @@
+ * @param objectPath The datapath for respective dataObject to be written out
+ * @param dataStructure The complex datastructure where *objectPath* datacontainer is stored
+ * //params with defaults
+ * @param delimiter The delimiter to be inserted into string
+ * @param includeIndex The boolean that determines if "Feature_IDs" are printed
+ * @param includeHeaders The boolean that determines if headers are printed
+ * @param componentsPerLine The amount of elements to be inserted before newline character
+ */
+void PrintSingleDataObject(std::ostream& outputStrm, const DataPath& objectPath, DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                           const std::string& delimiter, bool includeIndex, bool includeHeaders, size_t componentsPerLine)
+{
+  mesgHandler(IFilter::Message::Type::Info, fmt::format("Writing IArray ({}) to output stream", objectPath.getTargetName()));
+
+  auto* dataArray = dataStructure.getDataAs<IDataArray>(objectPath);
+  if(dataArray != nullptr)
+  {
+    ExecuteDataFunction(PrintDataArray{}, dataArray->getDataType(), outputStrm, dataArray, mesgHandler, shouldCancel, delimiter, componentsPerLine);
+  }
+  auto* stringArray = dataStructure.getDataAs<StringArray>(objectPath);
+  if(stringArray != nullptr)
+  {
+    PrintStringArray(outputStrm, *stringArray, mesgHandler, shouldCancel, delimiter, componentsPerLine);
+  }
+  auto* neighborList = dataStructure.getDataAs<INeighborList>(objectPath);
+  if(neighborList != nullptr)
+  {
+    ExecuteNeighborFunction(PrintNeighborList{}, neighborList->getDataType(), outputStrm, neighborList, mesgHandler, shouldCancel, delimiter, includeIndex, includeHeaders);
+  }
+};
+
+/**
+ * @brief [Single Output][Custom OStream] | writes out multiple arrays to ostream
+ * @param outputStrm The already opened output string to write to
+ * @param objectPaths The vector of datapaths for respective dataObjects to be written out
+ * @param dataStructure The complex datastructure where *objectPaths* datacontainers are stored
+ * @param mesgHandler The handler to send progress updates to
+ * @param shouldCancel The atomic boolean that determines cancel
+ * //params with defaults
+ * @param delimiter The delimiter to be inserted into string
+ * @param includeIndex The boolean that determines if "Feature_IDs" are printed
+ * @param includeHeaders The boolean that determines if headers are printed
+ * @param componentsPerLine The amount of elements to be inserted before newline character
+ * @param neighborLists The list of dataPaths of neighborlists to include
+ */
+void PrintDataSetsToSingleFile(std::ostream& outputStrm, const std::vector<DataPath>& objectPaths, DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler,
+                               const std::atomic_bool& shouldCancel, const std::string& delimiter, bool includeIndex, bool includeHeaders, size_t componentsPerLine,
```

Comment:
componentsPerLine is never used. Maybe we should remove it from the argument list